odoo/orm/domains.py

```diff
@@ -1111,6 +1111,25 @@ def _optimize_in_required(condition, model):
     return DomainCondition(condition.field_expr, condition.operator, value)
 
 
+@operator_optimization(['any', 'not any'], OptimizationLevel.FULL)
+def _optimize_bypass_comodel_id_lookup(condition, model):
+    """Avoid comodel's subquery, if it can be compared with the field directly"""
+    field = condition._field(model)
+    if (
+        model.env.is_superuser()
+        and '.' not in condition.field_expr
+        and field.relational
+        and not field.auto_join
+        and isinstance(subdomain := condition.value, DomainCondition)
+        and subdomain.field_expr == 'id'
+        and (op := subdomain.operator) in ('in', 'not in')
+    ):
+        if condition.operator == 'not any':
+            op = _INVERSE_OPERATOR[op]
+        return DomainCondition(condition.field_expr, op, subdomain.value)
```

Comment:
`a any (id in [False, 1])` should become `a in [1]` because "any" means "exists". These are probably removed by other optimizations, but there is no guarantee; so you should remove False from the values.

```suggestion
        return DomainCondition(condition.field_expr, op, [v for v in subdomain.value if v])
```